tests/integration/pilot/workloadentry_test.go

```diff
@@ -0,0 +1,141 @@
+//go:build integ
+// +build integ
+
+// Copyright Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pilot
+
+import (
+	"bytes"
+	"fmt"
+	"os/exec"
+	"testing"
+
+	"istio.io/istio/pkg/test/echo/common/scheme"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/echo"
+	"istio.io/istio/pkg/test/framework/components/echo/check"
+	"istio.io/istio/pkg/test/framework/components/echo/common/deployment"
+	"istio.io/istio/pkg/test/framework/resource/config/apply"
+)
+
+func TestWorkloadEntry(t *testing.T) {
+	// nolint: staticcheck
+	framework.NewTest(t).
+		Features("traffic.reachability").
+		Run(func(t framework.TestContext) {
+			// Define a simulated EW gateway
+			gatewayCfg := `apiVersion: networking.istio.io/v1alpha3
+kind: Gateway
+metadata:
+  name: ingress-ew
+  namespace: istio-system
+spec:
+  selector:
+    app: istio-ingressgateway
+  servers:
+  - port:
+      number: 8443
+      name: https
+      protocol: TLS
+    hosts:
+    - serviceentry.mesh.cluster.local
+    tls:
+      mode: AUTO_PASSTHROUGH
+`
+			// Setup a Gateway to act as simulated EW
+			if err := t.ConfigIstio().YAML("istio-system", gatewayCfg).Apply(apply.CleanupConditionally); err != nil {
+				t.Fatal(err)
+			}
+
+			// get pod IP of simulated EW gateway so we can use that IP in as the WorkloadEntry address
+			externalGetPodCommand := exec.Command("kubectl", "get", "pod", "-A", "-l", "app=istio-ingressgateway", "-o", `jsonpath="{.items[0].status.podIP}"`)
+			stdout := &bytes.Buffer{}
+			stderr := &bytes.Buffer{}
+			externalGetPodCommand.Stdout = stdout
+			externalGetPodCommand.Stderr = stderr
+			err := externalGetPodCommand.Run()
+			if err != nil {
+				t.Fatal(err)
+			}
+			ingressPodIP := stdout.String()
+
+			workloadEntryYaml := fmt.Sprintf(`apiVersion: networking.istio.io/v1beta1
+kind: ServiceEntry
+metadata:
+  name: a-se
+spec:
+  addresses:
+  - 240.240.34.56
+  hosts:
+  - serviceentry.mesh.cluster.local
+  ports:
+  - name: http
+    number: 80
+    protocol: HTTP
+    targetPort: 8080
+  location: MESH_INTERNAL
+  resolution: STATIC
+  workloadSelector:
+    labels:
+      app: a
+---
+apiVersion: networking.istio.io/v1beta1
+kind: WorkloadEntry
+metadata:
+  name: a-we
+  labels:
+    security.istio.io/tlsMode: istio
+spec:
+  network: other
+  ports:
+    http: 8443
+  address: %s
```

Comment:
On one hand, I like that you are adding a test of your use cases so that the project can ensure you are not broken etc.

On the other hand, this completely violates the model in Istio which assumes that an (IP,Network) uniquely identifies a workload. In this case, we have the same IP,Network for all workloads.

Reply:
We've taken some time to consider your feedback, investigate alternatives and run POCs of those alternatives.  Our conclusion was there's going to be some work involved in adopting a more idiomatic multi-network approach which will require other upstream contributions first. That's under way (design stage) and we are looking to deprecate this way of doing things in the future.

Do you think this test can be accepted while we work on the new implementation with the caveat that accepting the test doesn't mean Istio promises to support the use case? We'll look to remove the test in the future once we're no longer using Istio in this way.